framework/include/reporters/Reporter.h

```diff
@@ -95,7 +101,43 @@ class Reporter : public OutputInterface
   T & declareValueByName(const ReporterValueName & value_name, ReporterMode mode, Args &&... args0);
   ///@}
 
+  /**
+   * Declare a dummy value with type T.
+   *
+   * This is useful when you have a reporter that has optional values. In this case,
+   * you want to create references to all reporter values. However, because some values
+   * are optional, you need _something_ to fill into the reference. This helper will
+   * create a dummy value. It also allows for the passing of arguments in the case
+   * that your value is not trivially default constructable (constructable by default
+   * without arguments).
+   */
+  template <typename T, typename... Args>
+  T & declareDummyValue(Args &&... args);
```

Comment:
Is there a name we can use instead of "Dummy"? Proxy, Unused, Null, Empty. This is something we expect people to use, so I want the name to be a bit more professional.

Reply:
I like unused.